manim/constants.py

```diff
@@ -2,140 +2,141 @@
 (Use comma separated list for multiple entries)
 Choice(s): """
-INVALID_NUMBER_MESSAGE = "Invalid scene numbers have been specified. Aborting."
-NO_SCENE_MESSAGE = """
+INVALID_NUMBER_MESSAGE: str = "Invalid scene numbers have been specified. Aborting."
+NO_SCENE_MESSAGE: str = """
    There are no scenes inside that module
 """
 
 # Cairo and Pango stuff
-NORMAL = "NORMAL"
-ITALIC = "ITALIC"
-OBLIQUE = "OBLIQUE"
-BOLD = "BOLD"
+NORMAL: str = "NORMAL"
+ITALIC: str = "ITALIC"
+OBLIQUE: str = "OBLIQUE"
+BOLD: str = "BOLD"
 # Only for Pango from below
-THIN = "THIN"
-ULTRALIGHT = "ULTRALIGHT"
-LIGHT = "LIGHT"
-SEMILIGHT = "SEMILIGHT"
-BOOK = "BOOK"
-MEDIUM = "MEDIUM"
-SEMIBOLD = "SEMIBOLD"
-ULTRABOLD = "ULTRABOLD"
-HEAVY = "HEAVY"
-ULTRAHEAVY = "ULTRAHEAVY"
+THIN: str = "THIN"
+ULTRALIGHT: str = "ULTRALIGHT"
+LIGHT: str = "LIGHT"
+SEMILIGHT: str = "SEMILIGHT"
+BOOK: str = "BOOK"
+MEDIUM: str = "MEDIUM"
+SEMIBOLD: str = "SEMIBOLD"
+ULTRABOLD: str = "ULTRABOLD"
+HEAVY: str = "HEAVY"
+ULTRAHEAVY: str = "ULTRAHEAVY"
 
 
 # Geometry: directions
-ORIGIN = np.array((0.0, 0.0, 0.0))
+ORIGIN: np.ndarray = np.array((0.0, 0.0, 0.0))
 """The center of the coordinate system."""
 
-UP = np.array((0.0, 1.0, 0.0))
+UP: np.ndarray = np.array((0.0, 1.0, 0.0))
 """One unit step in the positive Y direction."""
 
-DOWN = np.array((0.0, -1.0, 0.0))
+DOWN: np.ndarray = np.array((0.0, -1.0, 0.0))
 """One unit step in the negative Y direction."""
 
-RIGHT = np.array((1.0, 0.0, 0.0))
+RIGHT: np.ndarray = np.array((1.0, 0.0, 0.0))
 """One unit step in the positive X direction."""
 
-LEFT = np.array((-1.0, 0.0, 0.0))
+LEFT: np.ndarray = np.array((-1.0, 0.0, 0.0))
 """One unit step in the negative X direction."""
 
-IN = np.array((0.0, 0.0, -1.0))
+IN: np.ndarray = np.array((0.0, 0.0, -1.0))
 """One unit step in the negative Z direction."""
 
-OUT = np.array((0.0, 0.0, 1.0))
+OUT: np.ndarray = np.array((0.0, 0.0, 1.0))
 """One unit step in the positive Z direction."""
 
 # Geometry: axes
-X_AXIS = np.array((1.0, 0.0, 0.0))
-Y_AXIS = np.array((0.0, 1.0, 0.0))
-Z_AXIS = np.array((0.0, 0.0, 1.0))
+X_AXIS: np.ndarray = np.array((1.0, 0.0, 0.0))
+Y_AXIS: np.ndarray = np.array((0.0, 1.0, 0.0))
+Z_AXIS: np.ndarray = np.array((0.0, 0.0, 1.0))
 
 # Geometry: useful abbreviations for diagonals
-UL = UP + LEFT
+UL: np.ndarray = UP + LEFT
 """One step up plus one step left."""
 
-UR = UP + RIGHT
+UR: np.ndarray = UP + RIGHT
 """One step up plus one step right."""
 
-DL = DOWN + LEFT
+DL: np.ndarray = DOWN + LEFT
 """One step down plus one step left."""
 
-DR = DOWN + RIGHT
+DR: np.ndarray = DOWN + RIGHT
 """One step down plus one step right."""
 
 # Geometry
-START_X = 30
-START_Y = 20
+START_X: int = 30
+START_Y: int = 20
 
 # Default buffers (padding)
-SMALL_BUFF = 0.1
-MED_SMALL_BUFF = 0.25
-MED_LARGE_BUFF = 0.5
-LARGE_BUFF = 1
-DEFAULT_MOBJECT_TO_EDGE_BUFFER = MED_LARGE_BUFF
-DEFAULT_MOBJECT_TO_MOBJECT_BUFFER = MED_SMALL_BUFF
+SMALL_BUFF: float = 0.1
```

Comment:
Is it really necessary to have these type hints here?
I think when defining constants, it is clear what it is, isn't it?

Reply:
We know this when we view this file. Now for user, they don't know. And this is mainly done for editor support.

Reply:
okay, then I am fine with this!